src/binaryManager.ts

```diff
@@ -93,33 +100,76 @@ async function getConfiguredMirrordBinary(): Promise<string | null> {
 }
 
 /**
- * Downloads mirrord binary (if needed) and returns its path
- */
-export async function getMirrordBinary(): Promise<string> {
+ * Toggles auto-update of mirrord binary.
+ * Criteria for auto-update:
+ * - Auto-update is enabled by default
+ * - if mirrord binary path is mentioned in workspace settings, then that is used
+ * - if a version is specified, that version is downloaded
+ * - if auto-update is enabled, then latest supported version is downloaded
+ * - if auto-update is disabled, any local mirrord binary is used
+ * 
+ * @param extensionActivate If true, then a global state is set so that any changes to the workspace settings pertaining
+ * to mirrord binary auto-update will prompt the user to reload the window.
+ * @returns Path to mirrord binary
+*/
+export async function getMirrordBinary(extensionActivate?: boolean): Promise<string> {
     const configured = await getConfiguredMirrordBinary();
+
     if (configured) {
+        vscode.window.showInformationMessage(`Using mirrord binary specified in settings: ${configured}`);
         return configured;
     }
+    const latestVersion = await getLatestSupportedVersion(10000);
+    const autoUpdateConfigured = vscode.workspace.getConfiguration().get("mirrord.autoUpdate");
+    const extensionPath = getExtensionMirrordPath().fsPath;
 
-    let foundLocal = await getLocalMirrordBinary();
-    // timeout is 1s if we have alternative or 10s if we don't
-    let timeout = foundLocal ? 1000 : 10000;
-    const latestVersion = await getLatestSupportedVersion(timeout);
-
-    // See if maybe we have it installed already, in correct version.
-    const localMirrord = await getLocalMirrordBinary(latestVersion);
-    if (localMirrord) {
-        const api = new MirrordAPI(localMirrord);
-        const installedVersion = await api.getBinaryVersion();
-        if (installedVersion === latestVersion) {
-            return localMirrord;
+    // if extension is activating, then set a global state to what was read in the workspace settings
```

Comment:
Why?

Reply:
mainly because we want to prompt the user to reload the IDE to avoid session timing out mid session

Reply:
Why would a session time out when the user didn't reload the IDE?

Reply:
user specifies new binary version -> starts debug session -> binary is downloading -> debug session times out

IMO instead now any changes different from original workspace settings pertaining to mirrord require reload

Reply:
The feature for specifying a version is *very* niche and we only expect users to use it when we instruct them to, so the edge case you're describing doesn't really matter. I'd rather keep the code simple.

Reply:
okay i a removing it